providers/targeted_refresh.md

```diff
@@ -0,0 +1,180 @@
+## Targeted Refresh
+
+If you followed the [Creating a new provider](writing_a_new_provider.md) guide then you already have a provider with full refresh working.  Or maybe you're working on an existing provider that only has full refresh and you'd like to add targeted refresh to it.
+
+### What is targeted refresh?
+
+First we should get some technicalities out of the way.  All refreshes have a target, "full" refresh is when the target is the top-level EMS and "targeted" refresh is when the target is anything else (for example a single VM).
+
+The primary benefit of targeted refresh is speed.
+
+Even a moderately large provider will start to take minutes to collect and save all of the inventory.  Doing this simply to pick up a single new VM is extremely wasteful.  Also since only one refresh is able to run at a time due to concurrency concerns, if a full refresh takes 10 minutes and you queue up your refresh request right after it starts the "latency" to getting your change into the database could take almost 20 minutes.  By contrast fetching a single resource like a VM and related inventory is extremely fast.
+
+There are two main differences between targeted refresh and full refresh:
+
+1. What is collected
+2. How inventory is deleted
+
+Targeted refresh only fetches inventory from the provider for the targets requested.  This cuts down on API calls and uses less memory.
+
+As for deletions, the most significant difference with targeted refresh is the "scope".  A full refresh considers the entire association (e.g.: `ems.vms.all`) to be in scope, so if something is in the association but not in the collected inventory it will be deleted.
+
+Targeted refresh requires scope to be limited, otherwise everything besides what you targeted would be deleted.  Most of this is abstracted away from the provider author but it is important to understand the fundamentals.  This is covered in way more detail in the [Provider Refresh](refresh.md) guide.
+
+### Getting Started
+
+The first thing to do is to set up the new classes that we need.  We'll need to add a `TargetCollection` class under the `Collector` and `Persister`.
+
+```ruby
+class ManageIQ::Providers::AwesomeCloud::Inventory::Collector::TargetCollection < ManageIQ::Providers::AwesomeCloud::Inventory::Collector
+end
+```
+
+```ruby
+class ManageIQ::Providers::AwesomeCloud::Inventory::Parser::TargetCollection < ManageIQ::Providers::AwesomeCloud::Inventory::Parser
+end
+```
+
+```ruby
+class ManageIQ::Providers::AwesomeCloud::Inventory::Persister::TargetCollection < ManageIQ::Providers::AwesomeCloud::Inventory::Persister
+  # Indicate that this is a targeted refresh
+  def targeted?
+    true
+  end
+
+  # This tells the Persister what to do with references, specifically this will
+  # lookup references for inventory objects which aren't in the collection which
+  # is required for targeted refresh.
+  def strategy
+    :local_db_find_missing_references
```

Comment:
NOTE: these strategy names are not very intuitive, wonder if we can default the strategy to this if `targeted?` is true so the provider author doesn't need to do this.